app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -29,24 +29,44 @@ class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
   # end
 
   def facebook
+    if Flipper.enabled?(:block_registration)
+      return redirect_to(new_user_session_path, alert: "Registration is currently blocked")
+    end
+
     generic_callback("facebook")
   end
 
   def google_oauth2
+    if Flipper.enabled?(:block_registration)
+      return redirect_to(new_user_session_path, alert: "Registration is currently blocked")
+    end
```

Comment:
Will this prevent social sign-in as well? Because the endpoint is the same, could you please validate the flow?

Reply:
i checked it, and yes It is preventing the signin functionality also , i am looking for a solution for this